c2corg_ui/static/js/apiservice.js

```diff
@@ -380,6 +387,39 @@ app.Api.prototype.updatePreferredLanguage = function(lang) {
 
 
 /**
+ * @param {string} type
+ * @param {undefined | string} token
+ * @return {!angular.$q.Promise<!angular.$http.Response>}
+ */
+app.Api.prototype.readFeed = function(type, token) {
+  var url = '';
+  var userId;
+
+  switch (type) {
```

Comment:
This switch is somehow a duplicate of the switch in the directive controller. As explained below, I would recommend to only have a "isProfile" boolean parameter. If ``true`` this function would use ``/profile-feed``, else it would test if the user is authenticated to choose among the standard or personal feeds.